dynet/expr.h

```diff
@@ -103,7 +125,23 @@ Expression f(const T& xs, const T1& arg1) {
   for (auto xi = xs.begin(); xi != xs.end(); ++xi) xis[i++] = xi->i;
   return Expression(pg, pg->add_function<F>(xis, arg1));
 }
+template <typename F, typename T>
```

Comment:
Give this a more descriptive name. Also it might be good to comment, unless what the function is doing is obvious from the name.